extensions/gmsa-dc/v1/Promote-DC.ps1

```diff
@@ -0,0 +1,51 @@
+###################################################################################
+# This script will automate the creation of a Forest, Domain, and Domain Controller
+#
+# This is only for automated e2e testing.  DO NOT use this for production.
+# Jeremy Wood (JeremyWx)
+# Version: 1.0.0.0
+###################################################################################
+
+# Create a working directory and change to it
+mkdir C:\gmsa
+Set-Location -Path C:\gmsa
+
+# Uncomment line below for troubleshooting
+Start-Transcript -Path "C:\gmsa\Promote.txt"
+
+# Download gMSA Setup Script
+Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/JeremyWx/aks-engine/master/extensions/gmsa-dc/v1/Setup-gMSA.ps1 -OutFile Setup-gMSA.ps1
+
+#Install NuGet
+Install-PackageProvider -Name NuGet -Force
+
+#Create Local Administrator
+$admpassword = ( "K8s" + -join ((48..57) + (97..122) | Get-Random -Count 64 | ForEach-Object {[char]$_}) )
+$admpassword_secure = ( $admpassword | ConvertTo-SecureString -AsPlainText -Force)
+New-LocalUser -Name gmsa-admin -Password $admpassword_secure
+Add-LocalGroupMember -Group "Administrators" -Member gmsa-admin
+Start-Sleep -Seconds 5
+
+# Make Setup-gMSA run on next boot
+$Logon = "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon"
+$RunOnce = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnce"
+Set-ItemProperty $Logon "AutoAdminLogon" -Value "1" -type String
+Set-ItemProperty $Logon "AutoLogonCount" -Value "2" -type DWord

```

Comment:
Why 2? Not sure I understand this line

Reply:
The process requires 2 reboots.  First we create an admin user and add it to the local Administrators group.  We add some required roles to the server and then reboot to login as our new admin user on our new domain.  Next we add our admin user to the Enterprise Admins group so we can create the KDS root key.  Because group membership is added to your token during login, we need to reboot again so the DC will recognize the admin user as a member of the Enterprise Admins group.